packages/rest/src/resource.ts

```diff
@@ -116,16 +116,19 @@ function cloneKeyProperties(context: DecoratorContext, target: Model, resourceTy
       // filtering. NOTE: Check for name equality instead of function equality
       // to deal with multiple copies of core being used.
       ...keyProperty.decorators.filter((d) => d.decorator.name !== $visibility.name),
-      {
-        decorator: $path,
-        args: [],
-      },
       {
         decorator: $resourceTypeForKeyParam,
         args: [{ node: target.node, value: resourceType, jsValue: resourceType }],
       },
     ];
 
+    if (!keyProperty.decorators.some((d) => d.decorator.name === $path.name)) {
```

Comment:
I'm sorry I didn't make the background clear. We are going to release SDK for some service this month. All the information SDK generator gets is from TCGC. Therefore we also expect the outcome from TCGC is correct too.

For this specific scenario, TCGC fully honor the information from [`getHttpOperation`](https://github.com/microsoft/typespec/blob/7e344d4ae6af65297adaac7b874218e703bbbdfa/packages/http/src/operations.ts#L36), so I assume we need change in http package. 

Could you try to call this function to make sure the `allowReserved` property of corresponding parameters also has correct output?

Reply:
are you algo observing an error on the tgcc area? could you please share how are you reproducing the tgcc error?

Reply:
I verified the fix with tcgc emitter and it works as expected 
<img width="524" alt="image" src="https://github.com/user-attachments/assets/6873cd57-83e8-4e26-a678-878a4da3b22e" />